test/Scrutor.Tests/ScanningTests.cs

```diff
@@ -140,7 +175,7 @@ public void AutoRegisterAsMatchingInterface()
                     .AsMatchingInterface()
                     .WithTransientLifetime());
 
-            Assert.Equal(2, Collection.Count);
+            Assert.Equal(3, Collection.Count);

```

Comment:
> Does register as self makes sense when you have an interface on it?

I'd argue; no. But people do the weirdest of stuff, so I wouldn't be surprised if this was something that someone would ask for.

I think I'd take a change where we just register as self _and_ all implemented interfaces. As I said, I think registration is cheap. It's resolution that costs (in terms of CPU cycles, not memory). If this is unacceptable to you, you can easily opt-out by explicitly specifying the services you want it registered as.